packages/go_router/lib/src/route.dart

```diff
@@ -271,6 +239,81 @@ class GoRoute {
   Map<String, String> extractPathParams(RegExpMatch match) =>
       extractPathParameters(_pathParams, match);
 
+  static String? _emptyRedirect(GoRouterState state) => null;
+}
+
+/// A route that is displayed visually above the matching parent route using the
+/// [Navigator].
+///
+/// The widget returned by [builder] is wrapped in [Page] and provided to the
+/// root Navigator, the nearest ShellRoute ancestor's Navigator, or the
+/// Navigator with a matching [parentNavigatorKey].
+///
+/// The Page will be either a [MaterialPage] (for [MaterialApp]),
+/// [CupertinoPage] (for [CupertinoApp], or [NoTransitionPage] (for
+/// [WidgetsApp]) depending on the application type.
```

Comment:
What do you think of this?

```suggestion
/// The Page depends on the application type: [MaterialPage] for
/// [MaterialApp], [CupertinoPage] for [CupertinoApp], or
/// [NoTransitionPage] for [WidgetsApp].
```